Add a “SELECT ALL” button in the `<BulkActionsToolbar>` (#10367)
marmelab/react-admin: docs/Buttons.md

To override the style of all instances of `<FilterButton>` using the [application-wide style overrides](./AppTheme.md#theming-individual-components), use the `RaFilterButton` key.

## `<SelectAllButton>`

> This button is an internal component used by react-admin in [the Datagrid default `bulkActionsToolbar`](./Datagrid.md#bulkactionstoolbar).

Reviewer: ```suggestion
```

> The `<SelectAllButton>` component is a button that is used to select all items from a resource, no matter the pagination.

Reviewer: ```suggestion
The `<SelectAllButton>` component allows users to select all items from a resource, no matter the pagination.
```

![SelectAllButton](./img/SelectAllButton.png)

### `label`

By default, `<SelectAllButton>` renders with the "Select all" (translated if the user locale isn't English), whether used on a creation or edition form. You can pass another label:

```jsx
const PostSelectAllButton = () => <SelectAllButton label="Select all posts" />;
```

**Tip**: The label will go through [the `useTranslate` hook](./useTranslate.md), so you can use translation keys.

### `limit`

By default, `<SelectAllButton>` select the 250 first items of your list. To customize this limit you can use the prop `limit` as follows:

```jsx
const PostSelectAllButton = () => <SelectAllButton limit={100} />;
```

### `queryOptions`

> `<SelectAllButton>` calls a `get` method of your `dataProvider` via a react-query's hook. You can customize the options you pass to this hook, e.g. to pass [a custom `meta`](./Actions.md#meta-parameter) to the call.

Author: The `SelectAllButton` doesn't necessary call the `getList` function. It can call the `getManyReference` if we are in a `ReferenceManyField`
